From 52acfecc48ad16ba1b7529d4f950a64ea596e8d9 Mon Sep 17 00:00:00 2001
From: Harsh J <harsh@cloudera.com>
Date: Thu, 4 Dec 2014 03:16:08 +0530
Subject: [PATCH 1041/2848] YARN-2891. Failed Container Executor does not
 provide a clear error message. Contributed by
 Dustin Cote. (harsh)

(cherry picked from commit a31e0164912236630c485e5aeb908b43e3a67c61)

Change-Id: I80aa674bf7551f974875f26801c7f60a0dc19657
---
 .../container-executor/impl/container-executor.c   |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/container-executor.c b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/container-executor.c
index 6d2fc09..11fba6d 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/container-executor.c
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/container-executor.c
@@ -550,7 +550,7 @@ int check_dir(const char* npath, mode_t st_mode, mode_t desired, int finalCompon
     int filePermInt = st_mode & (S_IRWXU | S_IRWXG | S_IRWXO);
     int desiredInt = desired & (S_IRWXU | S_IRWXG | S_IRWXO);
     if (filePermInt != desiredInt) {
-      fprintf(LOGFILE, "Path %s does not have desired permission.\n", npath);
+      fprintf(LOGFILE, "Path %s has permission %o but needs permission %o.\n", npath, filePermInt, desiredInt);
       return -1;
     }
   }
-- 
1.7.9.5

